lib/common/memory/src/mmap_type.rs

```diff
@@ -165,7 +165,10 @@ where
         Box::new({
             let mmap = self.mmap.clone();
             move || {
-                mmap.flush()?;
+                // flushing a zero-sized mmap can cause panicking on some systems
+                if mmap.len() > 0 {
+                    mmap.flush()?;
+                }
                 Ok(())
             }
         })
```

Comment:
Can we move the length check outside of the box, and return an empty closure if it's empty?

That may be a bit nicer so we don't clone the mmap structure (even though it's empty).

Reply:
Is it ensured that a mmap won't grow its size? just thinking that this returns a closure which will be run at a later time